### PR TITLE
cDNA and protein changes displayed in institute causatives table

### LINKED ISSUE #3259: Add cDNA and protein changes to institute causatives page
Important for cancer cases!

### PR DESCRIPTION
**Pull request by northwestwitch**
- Fix #3259  --> Display cDNA and prot changes in causatives table
- Moved causatives code from causatives views to a dedicated controller

Before:

image

After:

image

<details>
<summary>Testing on cg-vm1 server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. Make sure the PR is pushed and available on [Docker Hub](https://hub.docker.com/repository/docker/clinicalgenomics/scout-server-stage)
1. `ssh <USER.NAME>@cg-vm1.scilifelab.se`
1. `sudo -iu hiseq.clinical`
1. `ssh localhost`
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `podman ps`
1. Stop the service with current deployed branch: `systemctl --user stop scout.target`
1. Start the scout service with the branch to test: `systemctl --user start scout@<this_branch>`
1. Make sure the branch is deployed: `systemctl --user status scout.target`
</details>

**How to test**:
1. Pick and institute and go to "Causatives" page

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
- [x] code approved by DN
- [x] tests executed by DN, CR

### CODE REVIEW

File: scout/server/blueprints/institutes/controllers.py [hunk 1]
```diff
@@ -24,6 +30,66 @@
 TRACKS = {None: "Rare Disease", "rare": "Rare Disease", "cancer": "Cancer"}
 
 
+def causatives(institute_obj, request):
```
Comment (northwestwitch) [inline scout/server/blueprints/institutes/controllers.py:33]: Basically all this code is moved here from the causatives view

File: scout/server/blueprints/institutes/controllers.py [hunk 2]
```diff
@@ -24,6 +30,66 @@
...
+    if variants:
+        variants = sorted(
+            variants,
+            key=lambda k: k.get("hgnc_symbols", [None])[0] or k.get("str_repid") or "",
+        )
+
+    all_variants = {}
+    all_cases = {}
+    for variant_obj in variants:
+        update_representative_gene(variant_obj, variant_obj.get("genes", []))
```
Comment (northwestwitch) [inline scout/server/blueprints/institutes/controllers.py:63]: This is the only new line of code, and it is used to collect the hgvs_identifier and hgvsp_identifier shown on the table

**Review by dnil — COMMENTED** [review]
Looks great! As you noted yourself, there are some question marks regarding why there are so few HGVS_c populated on stage. Could be some old format, but worth looking into a bit before deploying - could also be an issue with transcript population / selecting representative genes!

**Comment by northwestwitch** [comment]
@dnil it should be fine now. I've re-reployed on stage the latest changes as well and most of the SNVs have cDNA changes now!

File: scout/server/blueprints/institutes/controllers.py [hunk 3]
```diff
@@ -24,6 +30,59 @@
...
+    if variants:
+        variants = sorted(
+            variants,
+            key=lambda k: k.get("hgnc_symbols", [None])[0] or k.get("str_repid") or "",
+        )
+
+    all_variants = {}
+    all_cases = {}
+    for variant_obj in variants:
+        if variant_obj["category"] in ["snv", "cancer"]:
```
Comment (northwestwitch) [inline scout/server/blueprints/institutes/controllers.py:63]: Collect additional values only if var is SNV or cancer var

**Review by dnil — APPROVED** [review]

**Comment by dnil** [comment]
✅ Tested on stage!

**Pull request merged**